liquid-rust-typeck/src/lib.rs

```diff
@@ -112,9 +118,17 @@ impl FixpointCtxt {
         did: DefId,
         body_span: Span,
         constraint: fixpoint::Constraint<TagIdx>,
+        qualifiers: Vec<ty::Qualifier>,
     ) -> Result<(), ErrorReported> {
         let kvars = self.kvars.into_fixpoint();
-        let task = fixpoint::Task::new(kvars, constraint);
+
+        let mut fx_qualifiers = Vec::new();
+
+        for qualifier in qualifiers {
+            fx_qualifiers.push(qualifier_into_fixpoint(&qualifier));
+        }
```

Comment:
Prefer a functional style, this allows you to reuse names.

```suggestion
        let qualifiers = qualifier.iter().map(qualifier_into_fixpoint).collect()
```